internal/common_context.go

```diff
@@ -151,7 +157,13 @@ type QueryParamInvalidTypeError struct {
 }
 
 func (e QueryParamInvalidTypeError) Error() string {
-	return fmt.Errorf("param %s=%s is not of type %s: %w", e.ParamName, e.ParamValue, e.ExpectedType, e.Err).Error()
+	return fmt.Errorf("query param %s=%s is not of type %s: %w", e.ParamName, e.ParamValue, e.ExpectedType, e.Err).Error()
+}
+
+func (e QueryParamInvalidTypeError) StatusCode() int { return 422 }
```

Comment:
Here an everywhere,I would have used existing constants

```suggestion
func (e QueryParamInvalidTypeError) StatusCode() int { return http.StatusUnprocessableEntity }
```

Same for 404 => http.StatusNotFound